src/MyApplication.cpp

```diff
@@ -1,67 +1,109 @@
+
+#include <gtkmm/application.h>
+#include <gtkmm/window.h>
+#include <gtkmm/glarea.h>
 
 using namespace Magnum;
 
-class MyApplication: public QOpenGLWidget {
-    public:
-        explicit MyApplication(Platform::GLContext& context, QWidget* parent = nullptr, Qt::WindowFlags f = nullptr);
+class MagnumWidget : public Gtk::GLArea {
+public:
+    explicit MagnumWidget(Platform::GLContext &context);
 
-    private:
-        void initializeGL() override;
-        void paintGL() override;
+private:
+    void onRealize();
+    bool onRender(const Glib::RefPtr<Gdk::GLContext> &context);
+    void onResize(int width, int height);
+    void onUnrealize();
 
-        Platform::GLContext& _context;
+    Platform::GLContext &_context;
 };
 
-MyApplication::MyApplication(Platform::GLContext& context, QWidget* parent, Qt::WindowFlags f): QOpenGLWidget{parent, f}, _context(context) {
-    /* TODO: Add your context format setup code here */
+MagnumWidget::MagnumWidget(Platform::GLContext &context) : Gtk::GLArea(), _context(context) {
+
+    /* Automatically re-render everything every time it needs to be drawn */
+    set_auto_render();
+
+    /* Set size requests and scaling behavior */
+    set_hexpand();
+    set_vexpand();
+    set_halign(Gtk::ALIGN_FILL);
+    set_valign(Gtk::ALIGN_FILL);
+    set_size_request(500, 500);
+
+    /* Set desired OpenGL version */
+    set_required_version(4, 5);
+
+    /* Connect signals to their respective handlers */
+    signal_realize().connect(sigc::mem_fun(this, &MagnumWidget::onRealize));
+    signal_render().connect(sigc::mem_fun(this, &MagnumWidget::onRender));
+    signal_resize().connect(sigc::mem_fun(this, &MagnumWidget::onResize));
+    signal_unrealize().connect(sigc::mem_fun(this, &MagnumWidget::onUnrealize));
 }
 
-void MyApplication::initializeGL() {
+void MagnumWidget::onRealize() {
+
+    /* Make sure the OpenGL context is current then configure it */
+    make_current();
     _context.create();
 
     /* TODO: Add your initialization code here */
+
 }
 
-void MyApplication::paintGL() {
-    /* Reset state to avoid Qt affecting Magnum */
-    GL::Context::current().resetState(GL::Context::State::ExitExternal);
+bool MagnumWidget::onRender(const Glib::RefPtr<Gdk::GLContext> &context) {
 
-    /* Using framebuffer provided by Qt as default framebuffer */
-    auto qtDefaultFramebuffer = GL::Framebuffer::wrap(defaultFramebufferObject(), {{}, {width(), height()}});
+    /* Retrieve the ID of the relevant framebuffer */
+    GLint framebufferID;
+    glGetIntegerv(GL_FRAMEBUFFER_BINDING, &framebufferID);
 
-    qtDefaultFramebuffer.clear(GL::FramebufferClear::Color);
+    /* Attache Magnum's framebuffer manager to the framebuffer provided by Gtkmm */
+    auto gtkmmDefaultFramebuffer = GL::Framebuffer::wrap(framebufferID, {{}, {get_width(), get_height()}});
+
+    /* Reset state to avoid Gtkmm affecting Magnum */
+    GL::Context::current().resetState(GL::Context::State::ExitExternal);
```

Comment:
I think this line should be the first thing in the function, to be safe. It won't affect the `glGet` call in any way.